lib/azure/armrest/armrest_service.rb

```diff
@@ -102,52 +55,15 @@ def fetch_token
       # the new portal or the New-AzureRoleAssignment powershell command.
       #
       def self.configure(options)
-        configuration = ArmrestConfiguration.new
-
-        options.each do |k,v|
-          configuration[k] = v
-        end
-
-        unless configuration.client_id && configuration.client_key
-          raise ArgumentError, "client_id and client_key must be specified"
-        end
-
-        configuration.api_version     ||= '2015-01-01'
-        configuration.grant_type      ||= 'client_credentials'
-        configuration.content_type    ||= 'application/json'
-        configuration.accept          ||= 'application/json'
-        configuration.subscription_id ||= fetch_subscription_id(configuration)
-
-        configuration
-      end
-
-      def self.fetch_subscription_id(config)
-        return @@subscriptions[config.as_cache_key] if @@subscriptions.has_key?(config.as_cache_key)
-
-        url = File.join(Azure::Armrest::RESOURCE, "subscriptions?api-version=#{config.api_version}")
-
-        response = rest_get(
-          url,
-          :content_type  => config.content_type,
-          :authorization => config.token
-        )
-
-        hash = JSON.parse(response)["value"].first
-
-        raise ArgumentError, "No associated subscription found" if hash.empty?
-
-        id = hash.fetch("subscriptionId")
-        @@subscriptions[config.as_cache_key] = id
+        Azure::Armrest::Configuration.new(options)
```

Comment:
Should we document this factory method obsolete and update your wiki to guide users to directly create a configuration object?